contracts/tokenlock/TokenLock.sol

```diff
@@ -0,0 +1,393 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.16;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "./OwnableInitializable.sol";
+import "./MathUtils.sol";
+import "./ITokenLock.sol";
+
+// this contract is based on GraphTokenLock
+// see https://github.com/graphprotocol/token-distribution/blob/main/contracts/GraphTokenLock.sol
+
+/**
+ * @title HatTokenLock
+ * @notice Contract that manages an unlocking schedule of tokens.
+ * @dev The contract lock manage a number of tokens deposited into the contract to ensure that
+ * they can only be released under certain time conditions.
+ *
+ * This contract implements a release scheduled based on periods and tokens are released in steps
+ * after each period ends. It can be configured with one period in which case it is like a plain TimeLock.
+ * It also supports revocation to be used for vesting schedules.
+ *
+ * The contract supports receiving extra funds than the managed tokens ones that can be
+ * withdrawn by the beneficiary at any time.
+ *
+ * A releaseStartTime parameter is included to override the default release schedule and
+ * perform the first release on the configured time. After that it will continue with the
+ * default schedule.
+ */
+// solhint-disable-next-line indent
+abstract contract TokenLock is OwnableInitializable, ITokenLock {
```

Comment:
why don't we use openzeppelin tockenLock?

Reply:
we inherited this and we have no reason to change to OZ now